packages/server/lib/browsers/firefox.ts

```diff
@@ -546,40 +509,116 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
     await fs.writeFile(path.join(profileDir, 'chrome', 'userChrome.css'), userCss)
   }
 
-  launchOptions.args = launchOptions.args.concat([
-    '-profile',
-    profile.path(),
-  ])
-
-  debug('launch in firefox', { url, args: launchOptions.args })
-
-  const browserInstance = launch(browser, 'about:blank', remotePort, launchOptions.args, {
-    // sets headless resolution to 1280x720 by default
-    // user can overwrite this default with these env vars or --height, --width arguments
+  // resolution of exactly 1280x720
+  // (height must account for firefox url bar, which we can only shrink to 1px ,
+  // and the total size of the window url and tab bar, which is 85 pixels for a total offset of 86 pixels)
+  const BROWSER_ENVS = {
+    MOZ_REMOTE_SETTINGS_DEVTOOLS: '1',
     MOZ_HEADLESS_WIDTH: '1280',
-    MOZ_HEADLESS_HEIGHT: '721',
+    MOZ_HEADLESS_HEIGHT: '806',
     ...launchOptions.env,
+  }
+
+  debug('launching geckodriver with browser envs %o', BROWSER_ENVS)
+
+  // create the geckodriver process, which we will use WebDriver Classic to open the browser
+  const geckoDriverInstance = await GeckoDriver.create({
+    host: '127.0.0.1',
+    port: geckoDriverPort,
+    marionetteHost: '127.0.0.1',
+    marionettePort,
+    webdriverBidiPort: webDriverBiDiPort,
+    profilePath: profile.path(),
+    binaryPath: browser.path,
+    // To pass env variables into the firefox process, we CANNOT do it through capabilities when starting the browser.
+    // Since geckodriver spawns the firefox process, we can pass the env variables directly to geckodriver, which in turn will
+    // pass them to the firefox process
+    // @see https://bugzilla.mozilla.org/show_bug.cgi?id=1604723#c20 for more details
+    spawnOpts: {
+      stdio: ['ignore', 'pipe', 'pipe'],
+      env: {
+        ...BROWSER_ENVS,
+        ...process.env,
+      },
+    },
   })
 
+  const wdcInstance = new WebDriverClassic('127.0.0.1', geckoDriverPort)
+
+  debug('launch in firefox', { url, args: launchOptions.args })
+
+  const capabilitiesToSend = {
+    // browser capabilities are going here as exact
```

Comment:
I'm not sure what this comment is saying.

Reply:
I can probably just delete it

Reply:
deleted in fb54eaaa9a